### PR TITLE
#46: Unify move tool and resize tool to change-bounds tool with resize handles

### PR DESCRIPTION
**Pull request by martin-fleck-at**
- Provide action and command to (de-)activate element resize mode
- Add resize handle element to workflow model and views
- Active resizeable element has one resize handle in each corner
- Each resize handle allows changing the size freely
- Remove previous resizing methods (key, mouse wheel)
- Remove custom container to re-activate zoom mouse listener
- Track selection to properly disable resize on multi-select

change-bounds

### CODE REVIEW

File: client/packages/glsp-sprotty/src/features/tools/change-bounds-tool.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+import { forEachElement, getIndex, isSelectedBoundsAware } from "../../utils/smodel-util";
+import { isBoundsAwareMoveable } from "../change-bounds/model";
+import { ChangeBoundsOperationAction } from "../operation/operation-actions";
+import { isResizeable, ResizeHandleLocation, SResizeHandle } from "../resize/model";
+import { addResizeHandles, SwitchResizeModeAction } from "../resize/resize";
+import { FeedbackMoveMouseListener } from "../tool-feedback/move-tool-feedback";
+import { Tool } from "../tool-manager/tool";
+
+/**
+ * The change bounds tool has the license to move multiple elements or resize a single element by implementing the ChangeBounds operation.
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/tools/change-bounds-tool.ts]: I see what you did there

File: client/packages/glsp-sprotty/src/features/tools/change-bounds-tool.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+ * +-----------+------------------+----------------------------
+ * | Move      | MoveAction       | ChangeBoundsOperationAction
+ * | Resize    | SetBoundsAction  | ChangeBoundsOperationAction
+ *
+ * To provide a visual client updates during move we install the `FeedbackMoveMouseListener` and to provide visual client updates during resize
+ * and send the server updates we install the `ChangeBoundsMouseListener`.
+ */
+@injectable()
+export class ChangeBoundsTool implements Tool {
+
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/tools/change-bounds-tool.ts]: Unnecessary blank line

File: client/packages/glsp-sprotty/src/features/resize/resize.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ * Contributors:
+ *  Martin Fleck - initial API and implementation
+ ******************************************************************************/
+import { injectable } from "inversify";
+import { Action, Command, CommandExecutionContext, CommandResult, SModelElement, SModelRoot, SParentElement } from "sprotty/lib";
+import { isNotUndefined } from "../../utils/smodel-util";
+import { isResizeable, ResizeHandleLocation, SResizeHandle } from "./model";
+
+
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/resize/resize.ts]: Unnecessary blank line

File: client/packages/glsp-sprotty/src/features/tools/resize-tool.ts [hunk 4]
```diff
@@ -27,137 +29,176 @@ import { ExtendedKeyListener } from "./key-tool";
  */
 @injectable()
 export class ResizeTool implements Tool {
+
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/tools/resize-tool.ts]: Please remove this empty line

**Comment by martin-fleck-at** [comment]
@tortmayr Thank you for your review, I addressed the minor comments in a follow-up commit and will attack the two larger points (removal of ExtendedKeyListener and usage of TypeHintsService) in a follow-up PR.

**Comment by tortmayr** [comment]
LGTM

**Pull request merged**